test/secmemtest.c

```diff
@@ -75,12 +75,31 @@ static int test_sec_mem(void)
     TEST_ptr_null(OPENSSL_secure_malloc((size_t)-1));
     TEST_true(CRYPTO_secure_malloc_done());
 
+    /*-
+     * There was also a possible infinite loop when the number of
+     * elements was 1<<31, as |int i| was set to that, which is a
+     * negative number. However, it requires minimum input values:
+     *
+     * CRYPTO_secure_malloc_init((size_t)1<<34, (size_t)1<<4);
+     *
+     * Which really only works on 64-bit systems, and even then the
+     * code attempts to allocate 16 GB secure memory arena. Linux
+     * can deal with this better than other Unixy OS's (e.g. MacOS)
```

Comment:
Well, I'd say the test should be disabled, because you simply can't make an assumption that *every* system that the test will be executed at has that much memory [for example none of my workstations do, let alone IoT-kind], virtual or physical.

Otherwise another parameter is of course memorylocked limit, when attempt to lock more than is bound to fail. MacOS or Solaris don't have it, but Linux and BSD do. BTW, surpassing the limit is privileged operation, so that failure to lock 16GB is not an option for non-root user irregardless amount of available physical memory...